mvnxparser/xmlStreamReader/xmlStreamReader.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright: (C) 2017 iCub Facility
+ * Author: Diego Ferigo
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+
+/**
+ * @file xmlStreamReader.cpp
+ * @brief Validate and parse efficiently big XML documents
+ * @author Diego Ferigo
+ * @date 06/04/2017
+ */
+
+#include "xmlStreamReader.h"
+
+using namespace std;
+using namespace xmlstream;
+
+xmlStreamReader::xmlStreamReader()
+{
+    if (QCoreApplication::instance() == nullptr) {
+        dummyArgc = 1;
+        dummyArgv = new char[1]{' '};
+        coreApp   = new QCoreApplication(dummyArgc, &dummyArgv);
+    }
+}
+xmlStreamReader::xmlStreamReader(string documentFile) : xmlStreamReader()
+{
+    setDocument(documentFile);
+}
+
+xmlStreamReader::xmlStreamReader(string documentFile, string schemaFile)
+    : xmlStreamReader(documentFile)
+{
```

Comment:
Just a curiosity.
Usually the designated initialiser is the more complete one, not the contrary.
Why this choice here?